[Wolfi Package Request]: yamllint

### What software would you like us to add to wolfi-os. Ideally include a URL to the project and its source.

https://github.com/adrienverge/yamllint 

### which versions of the software should we include?

latest

### Add some justification for why this specific package and versions are important.

This is a yaml linter and it'll be useful as a yaml linter. 

I tried the following but I was not able to get this to working stage. 
```yaml
package:
  name: yamllint
  version: 1.33.0
  epoch: 1
  description: "A linter for YAML"
  copyright:
    - license: GPL-3.0-only
  dependencies
  # runtime:
  #   - python3:

environment:
  contents:
    packages:
      - build-base
      - busybox
      - ca-certificates-bundle
      - py3-pip
      - py3-setuptools
      - python3
      - wolfi-base

pipeline:
  - uses: git-checkout
    with:
      repository: https://github.com/adrienverge/yamllint
      tag: v${{package.version}}
      expected-commit: 8713140e99914c0cb6e07364daedfb140847c3c5

  - runs: |
      python3 setup.py build

  - runs: |
      python3 setup.py install --prefix=/usr --root="${{targets.destdir}}"

  - uses: strip

update:
  enabled: true
  github:
    identifier: adrienverge/yamllint
    use-tag: true
```

### Please check all that apply

- [X] This package has an un-restrictive license
- [X] The package/versions proposed are actively maintained upstream
- [ ] I am interested in adding this package to Wolfi OS myself
- [ ] I am willing to help maintain this package

Thanks for filing @kranurag7! I took a look at the melange yaml you provided and was able to make a few minor adjustments and get it to build and install successfully

```diff
 package:
   name: yamllint
   version: 1.33.0
-  epoch: 1
+  epoch: 0
   description: "A linter for YAML"
   copyright:
-    - license: GPL-3.0-only
-  dependencies
-  # runtime:
-  #   - python3:
+    - license: GPL-3.0
+  dependencies:
+    runtime:
+      - py3-yaml
+      - py3-pathspec

 environment:
   contents:
    packages:
      - build-base
      - busybox
      - ca-certificates-bundle
      - py3-pip
      - py3-setuptools
      - python3
      - wolfi-base

pipeline:
  - uses: git-checkout
    with:
      repository: https://github.com/adrienverge/yamllint
      tag: v${{package.version}}
      expected-commit: 8713140e99914c0cb6e07364daedfb140847c3c5

  - runs: |
      python3 setup.py build

  - runs: |
      python3 setup.py install --prefix=/usr --root="${{targets.destdir}}"

  - uses: strip

update:
  enabled: true
  github:
    identifier: adrienverge/yamllint
    use-tag: true
```

You can see I was able to add it and run it successfully after it was built with `make package/yamllint`

```sh
… dev/found-it/os [!?] with 🐃 took 30s
➜ cp yamllint.yaml packages/

… dev/found-it/os [!?] with 🐃
➜ make local-wolfi
docker run --rm -it \
                --mount type=bind,source="/Users/james/dev/found-it/os/packages",destination="/work/packages",readonly \
                --mount type=bind,source="/Users/james/dev/found-it/os/local-melange.rsa.pub",destination="/etc/apk/keys/local-melange.rsa.pub",readonly \
                --mount type=bind,source="/var/folders/zn/39yscsyx2r722z6yvdq3351r0000gn/T/tmp.yUJQMJdVM1/repositories",destination="/etc/apk/repositories",readonly \
                -w "/work/packages" \
                cgr.dev/chainguard/wolfi-base:latest
fb08f1ff90be:/work/packages# apk add yamllint
fetch https://packages.wolfi.dev/os/aarch64/APKINDEX.tar.gz
(1/17) Installing bzip2 (1.0.8-r4)
(2/17) Installing expat (2.5.0-r3)
(3/17) Installing libffi (3.4.4-r3)
(4/17) Installing gdbm (1.23-r4)
(5/17) Installing xz (5.4.5-r1)
(6/17) Installing libgcc (13.2.0-r4)
(7/17) Installing libstdc++ (13.2.0-r4)
(8/17) Installing mpdecimal (2.5.1-r3)
(9/17) Installing ncurses-terminfo-base (6.4_p20231125-r1)
(10/17) Installing ncurses (6.4_p20231125-r1)
(11/17) Installing readline (8.2-r3)
(12/17) Installing sqlite (3.40.0-r1)
(13/17) Installing python-3.12 (3.12.1-r0)
(14/17) Installing py3-pathspec (0.12.1-r0)
(15/17) Installing yaml (0.2.5-r1)
(16/17) Installing py3-yaml (6.0.1-r1)
(17/17) Installing yamllint (1.33.0-r0)
OK: 61 MiB in 31 packages

fb08f1ff90be:/work/packages# yamllint /work/packages/yamllint.yaml
/work/packages/yamllint.yaml
  1:1       warning  missing document start "---"  (document-start)

fb08f1ff90be:/work/packages#
```

Feel free to add it to wolfi or if you'd like me to I'd be more than happy!

--- a/yamllint.yaml
+++ b/yamllint.yaml
@@ -39,3 +39,4 @@ update:
   github:
     identifier: adrienverge/yamllint
     use-tag: true
+    strip-prefix: v

--- a/yamllint.yaml
+++ b/yamllint.yaml
@@ -7,8 +7,8 @@ package:
     - license: GPL-3.0
   dependencies:
     runtime:
-      - py3-yaml
       - py3-pathspec
+      - py3-yaml
 
 environment:
   contents:

--- a/yamllint.yaml
+++ b/yamllint.yaml
@@ -4,7 +4,7 @@ package:
   epoch: 0
   description: "A linter for YAML"
   copyright:
-    - license: GPL-3.0
+    - license: GPL-3.0-r-later
   dependencies:
     runtime:
       - py3-pathspec

--- a/yamllint.yaml
+++ b/yamllint.yaml
@@ -4,7 +4,7 @@ package:
   epoch: 0
   description: "A linter for YAML"
   copyright:
-    - license: GPL-3.0-r-later
+    - license: GPL-3.0-or-later
   dependencies:
     runtime:
       - py3-pathspec

--- a/yamllint.yaml
+++ b/yamllint.yaml
@@ -1,0 +1,41 @@
+package:
+  name: yamllint
+  version: 1.33.0
+  epoch: 0
+  description: "A linter for YAML"
+  copyright:
+    - license: GPL-3.0
+  dependencies:
+    runtime:
+      - py3-yaml
+      - py3-pathspec
+
+environment:
+  contents:
+    packages:
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - py3-pip
+      - py3-setuptools
+      - python3
+      - wolfi-base
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/adrienverge/yamllint
+      tag: v${{package.version}}
+      expected-commit: 8713140e99914c0cb6e07364daedfb140847c3c5
+
+  - uses: python/build
+
+  - uses: python/install
+
+  - uses: strip
+
+update:
+  enabled: true
+  github:
+    identifier: adrienverge/yamllint
+    use-tag: true

